src/context/AppContext.tsx

```diff
@@ -0,0 +1,81 @@
+import { createContext, ReactNode, useEffect, useMemo, useState } from 'react';
+
+import { Superhero } from '@interfaces/superhero.interface';
+import { fetchSuperheroes } from '@utils/api';
+import { useDebouncedSearch } from '@utils/hooks/useDebouncedSearch';
+import { useRandomItems } from '@utils/hooks/useRandomItems';
+import { useAsyncError } from '@utils/useAsyncError';
+
+export interface AppState {
+  areHeroesFetched: boolean;
+  areHeroesFetching: boolean;
+  displayedHeroes?: Superhero[];
+  hasSearchResults: boolean;
+  searchText?: string;
+  setSearchText: (searchText: string) => void;
+  superheroes?: Superhero[];
+}
+
+export const initialState: AppState = {
+  areHeroesFetched: false,
+  areHeroesFetching: true,
+  displayedHeroes: undefined,
+  hasSearchResults: false,
+  searchText: '',
+  setSearchText: () => {},
+  superheroes: undefined,
+};
+
+export const AppContext = createContext<AppState>(initialState);
+
+export const AppContextProvider = ({ children }: { children: ReactNode }) => {
```

Comment:
@kamil-borecki-cc yes, you're probably right. I just really wanted the random superheroes (or the data that was searched for) to be preserved between route changes. To achieve this, I needed a global state.

I agree though, that such state should belong to a specific Superhero feature. But I decided it's going to be okay to just use something like `AppContext`, since the whole app is basically all about Superheroes.

I'm pretty sure there are better solutions here, I'm open to discuss this subject sometime :)